triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -1118,9 +1147,35 @@ object Runner {
     }
   }
 
+  private def mapSize(smap: SValue): Int = {
+    smap.expect("SMap", { case SMap(_, values) => values.size }).orConverterException
+  }
+
+  private def numberOfActiveContracts(svalue: SValue, level: Trigger.Level): Option[Int] = {
+    level match {
+      case Trigger.Level.High =>
+        // The following code should be kept in sync with the ACS variant type in Internal.daml
+        // svalue: TriggerState s
+        val result = for {
+          acs <- svalue.expect("SRecord", { case SRecord(_, _, values) => values.get(0) })
+          activeContracts <- acs.expect("SRecord", { case SRecord(_, _, values) => values.get(0) })
+          size <- activeContracts.expect(
+            "SMap",
+            { case SMap(_, values) => values.values.map(mapSize).sum },
+          )
+        } yield size
+
+        Some(result.orConverterException)
+
+      case Trigger.Level.Low =>
+        None
```

Comment:
Low level triggers can have arbitrary client defined state, so it makes no sense to try and extract the ACS here

Reply:
For this type of scenario, maybe we can use information from the ledger transaction source to make estimates on the ACS size? That would then provide better insulation against ACS blow up in low level triggers.